packages/svelte-ux/src/lib/components/settings.ts

```diff
@@ -49,3 +57,147 @@ export function getFormatNumberOptions(style?: FormatNumberStyle) {
 
   return toRet;
 }
+
+export function getFormatDateOptions(options?: FormatDateOptions) {
+  // if custom is set && variant is not set, let's put custom as variant
+  const variant: FormatDateOptions['variant'] =
+    options?.custom && options?.variant === undefined ? 'custom' : options?.variant ?? 'default';
+
+  const settings = getSettings();
+
+  const baseParsing = options?.baseParsing ?? settings.formats?.dates?.baseParsing ?? 'MM/dd/yyyy';
+
+  const custom = options?.custom ?? '';
+
+  let toRet: {
+    locales: string;
+    baseParsing: string;
+    weekStartsOn: DayOfWeek;
+    variant: DateFormatVariant;
+    custom: CustomIntlDateTimeFormatOptions;
+    presets: {
+      days: Record<DateFormatVariant, CustomIntlDateTimeFormatOptions>;
+      weeks: Record<DateFormatVariant, CustomIntlDateTimeFormatOptions>;
+      months: Record<DateFormatVariant, CustomIntlDateTimeFormatOptions>;
+      monthsYears: Record<DateFormatVariant, CustomIntlDateTimeFormatOptions>;
+      years: Record<DateFormatVariant, CustomIntlDateTimeFormatOptions>;
+    };
+    ordinalSuffixes: Record<string, OrdinalSuffixes>;
+    dico: {
```

Comment:
>  assume dico is short for dictionary? I haven't seen this abbreviation if so. Could we call it dictionary or translation?

Oups, yes I'm use to use `dico`, but it's maybe not so common ^^.
I feel that `translation` is like: 
```ts
translation: {
  en: {
    "day": "Day"
  }
}
```
`dictionary` is more:
```ts
dictionary: {
  "day": "Day"
}
```
Saying that, it's maybe better to go to `translation` (more future proof? fallbacks, ...)

---

True, that I will need also `global` things (less urgent).

Is it better to start directly under another place beside `formats`? or later? or?
(I would like to not have a too big PR, but would like also to not break by moving things around 2 weeks after introducing them ^^)

Let me know

Reply:
TBH, I'm not an expert here.  I'm hoping with Svelte UX / LayerChart to push as much of this into `Intl.*` browser APIs or similar to not get in the way, but also not have to maintain it.

With that said, I know these kind of strings still need to be handled.  At first I thought using `settings()` (i.e. `context`) might be an issue with SSR (will we always render with English, but when it hydrates in the browser and switches to say French, will there be a flash, or can we still set French on the backend by reading `Accept-Language` or a user preference from the DB).  Based on the SvelteKit [docs](https://kit.svelte.dev/docs/state-management#using-stores-with-context), I don't think context will be an issue here, so maybe I'm overthinking it.

I assume to do i18n properly you would want to use something like [typesafe-i18n](https://github.com/ivanhofer/typesafe-i18n) or [Paraglide JS](https://inlang.com/m/gerre34r/library-inlang-paraglideJs) with some preprocessing, but I don't think that is something we would want as a dependency within this project...

All that's to say, I think this works for me, but I could see some additional iteration in the future, especially with people with more direct needs / understanding (we don't need to boil the ocean now).  My hope is to keep the number of translatable strings within Svelte UX and LayerChart under 10 :).

Reply:
> SSR
I didn't look so much into this (I will at some point). But maybe not in this PR. I think that it should be ok with the current setup, would b

Reply:
Let's use top-level `translations` for now to allow to grow (and now have to come up with a "namespace" for some things like button labels.

Reply:
Ok, I did something. It will be easy to add https://inlang.com/ in userland.
I added `Ok` and `Cancel`, just to demo. But I don't expect to expend here in this PR.

Reply:
Looks good, thanks!